bootable-jar/WFLY-14934_Server_Mode.adoc

```diff
@@ -0,0 +1,122 @@
+= Server Mode for Bootable JAR Maven Plugin
+:author:           Jean-Francois Denise
+:email:             jdenise@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+Introduce a mode that makes the Bootable JAR Maven Plugin to build a server installation instead of a Bootable JAR.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-14934[WFLY-14934]
+
+=== Related Issues
+
+* New WildFly S2I: https://issues.redhat.com/browse/WFLY-14936[WFLY-14936]
+
+* JKube evolved support: https://issues.redhat.com/browse/WFLY-14936[WFLY-14938]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* TBD
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [ ] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+* https://github.com/wildfly-extras/wildfly-jar-maven-plugin/[Bootable JAR Maven plugin]
+
+* https://github.com/eclipse/jkube[JKube]
+
+* https://github.com/wildfly/wildfly-s2i[WildFly s2i]
+
+* https://github.com/wildfly/quickstart[WildFly quickstarts]
+
+* ODO dev files.
+
+=== Relevant Installation Types
+// Remove the x next to the relevant field if the feature in question is not relevant
+// to that kind of WildFly installation
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [ ] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+Ability to build and run an un-packaged server. All the goals defined in the Maven plugin are applicable to a server produced inside an installation dir.
+
+=== Hard Requirements
+
+==== Plugin requirements
+
+* Plugin behave the same as when producing a Bootable JAR, the only visible difference is that the project target dir contains 
+   a directory (named server by default) that contains the server and any deployments.
```

Comment:
"any deployments". Should it be "any deployment artifacts generated by the Maven project"?

Reply:
We can have multiple deployments, the one generated in the project but also deployments added as dependencies.